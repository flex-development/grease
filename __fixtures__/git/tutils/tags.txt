tutils@6.0.0-alpha.24
tutils@6.0.0-alpha.23
tutils@6.0.0-alpha.22
tutils@6.0.0-alpha.21
tutils@6.0.0-alpha.20
tutils@6.0.0-alpha.19
tutils@6.0.0-alpha.18
tutils@6.0.0-alpha.17
tutils@6.0.0-alpha.16
tutils@6.0.0-alpha.15
tutils@6.0.0-alpha.14
tutils@6.0.0-alpha.13
tutils@6.0.0-alpha.12
tutils@6.0.0-alpha.11
tutils@6.0.0-alpha.10
tutils@6.0.0-alpha.9
tutils@6.0.0-alpha.8
tutils@6.0.0-alpha.7
tutils@6.0.0-alpha.6
tutils@6.0.0-alpha.5
tutils@6.0.0-alpha.4
tutils@6.0.0-alpha.3
tutils@6.0.0-alpha.2
tutils@6.0.0-alpha.1
tutils@5.0.1
tutils@5.0.0
tutils@5.0.0-alpha.1
tutils@5.0.0-dev.2
tutils@5.0.0-dev.1
tutils@4.9.0-dev.1
tutils@4.8.0
tutils@4.7.0
tutils@4.6.0
tutils@4.5.0
tutils@4.4.0
tutils@4.3.0
tutils@4.2.3
tutils@4.2.2
tutils@4.2.1
tutils@4.2.0
tutils@4.1.1
tutils@4.1.0
tutils@4.0.3
tutils@4.0.2
tutils@4.0.1
tutils@4.0.0
tutils@4.0.0-dev.0
tutils@3.1.7
tutils@3.1.6
tutils@3.1.5
tutils@3.1.4
tutils@3.1.3
tutils@3.1.2
tutils@3.1.1
tutils@3.1.0
tutils@3.0.0
tutils@2.0.0
tutils@1.0.0
