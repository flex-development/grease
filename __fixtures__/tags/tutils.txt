tag: tutils@6.0.0-alpha.22
















tag: tutils@6.0.0-alpha.21





tag: tutils@6.0.0-alpha.20















tag: tutils@6.0.0-alpha.19















tag: tutils@6.0.0-alpha.18

tag: tutils@6.0.0-alpha.17







tag: tutils@6.0.0-alpha.16



tag: tutils@6.0.0-alpha.15


















tag: tutils@6.0.0-alpha.14













tag: tutils@6.0.0-alpha.13











tag: tutils@6.0.0-alpha.12






tag: tutils@6.0.0-alpha.11



























































































































































































































































tag: tutils@6.0.0-alpha.10
























































































tag: tutils@6.0.0-alpha.9






tag: tutils@6.0.0-alpha.8






























tag: tutils@6.0.0-alpha.7

tag: tutils@6.0.0-alpha.6


tag: tutils@6.0.0-alpha.5





tag: tutils@6.0.0-alpha.4

tag: tutils@6.0.0-alpha.3

tag: tutils@6.0.0-alpha.2




tag: tutils@6.0.0-alpha.1














































tag: tutils@5.0.1










































































tag: tutils@5.0.0




tag: tutils@5.0.0-alpha.1




tag: tutils@5.0.0-dev.2







































tag: tutils@5.0.0-dev.1






tag: tutils@4.9.0-dev.1













































tag: tutils@4.8.0

tag: tutils@4.7.0

tag: tutils@4.6.0





tag: tutils@4.5.0




tag: tutils@4.4.0





tag: tutils@4.3.0

tag: tutils@4.2.3

tag: tutils@4.2.2



tag: tutils@4.2.1








tag: tutils@4.2.0






tag: tutils@4.1.1

tag: tutils@4.1.0












tag: tutils@4.0.3

tag: tutils@4.0.2









tag: tutils@4.0.1







tag: tutils@4.0.0







tag: tutils@4.0.0-dev.0










tag: tutils@3.1.7


tag: tutils@3.1.6


tag: tutils@3.1.5



tag: tutils@3.1.4


tag: tutils@3.1.3


tag: tutils@3.1.2


tag: tutils@3.1.1










































tag: tutils@3.1.0



tag: tutils@3.0.0







tag: tutils@2.0.0






tag: tutils@1.0.0




