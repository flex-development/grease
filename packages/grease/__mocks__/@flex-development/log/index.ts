/**
 * @file Node Module Mock - @flex-development/log
 * @module grease/tests/mocks/flex-development/log
 * @see https://jestjs.io/docs/next/manual-mocks#mocking-node-modules
 * @see https://github.com/flex-development/log
 */

export default jest.fn((...args) => {
  return jest.requireActual('@flex-development/log').default(...args)
})
