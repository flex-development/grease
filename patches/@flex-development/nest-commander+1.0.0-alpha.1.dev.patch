diff --git a/dist/providers/command-runner.service.mjs b/dist/providers/command-runner.service.mjs
index v1.0.0-alpha.1..v1.0.0-alpha.1 100644
--- a/dist/providers/command-runner.service.mjs
+++ b/dist/providers/command-runner.service.mjs
@@ -15,1 +15,1 @@
-import { cast, constant, define, entries, fallback, flat, get, isString, isUndefined, lowercase, pick, select } from "@flex-development/tutils";
+import { cast, constant, define, entries, fallback, flat, get, isString, isUndefined, lowercase, noop, pick, select } from "@flex-development/tutils";
@@ -76,1 +76,8 @@
-      env && option.env(env), choices.length && option.choices(choices), option.argParser(method.handler.bind(metadata.instance)), command.addOption(option);
+      env && option.env(env);
+      choices.length && option.choices(choices);
+      const { parseArg = noop } = option;
+      option.argParser((value, previous) => {
+        parseArg(value, previous)
+        return method.handler.bind(metadata.instance)(value, previous)
+      });
+      command.addOption(option);
