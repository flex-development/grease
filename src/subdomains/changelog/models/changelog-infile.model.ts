/**
 * @file Models - ChangelogInfile
 * @module grease/changelog/models/ChangelogInfile
 */

import { isFile } from '@flex-development/mlly'
import pathe from '@flex-development/pathe'
import { DOT, type Stringafiable } from '@flex-development/tutils'
import fs from 'node:fs'

/**
 * Changelog infile model.
 *
 * @class
 * @implements {Stringafiable}
 */
class ChangelogInfile implements Stringafiable {
  /**
   * Absolute path to infile.
   *
   * @public
   * @readonly
   * @instance
   * @member {string} path
   */
  public readonly path: string

  /**
   * Create a new changelog infile.
   *
   * @param {string?} [infile='CHANGELOG.md'] - Infile path or filename
   * @param {string?} [cwd=DOT] - Path to current working directory
   */
  constructor(infile: string = 'CHANGELOG.md', cwd: string = DOT) {
    if (!isFile(this.path = pathe.resolve(cwd, infile))) {
      throw new Error('invalid infile', { cause: { infile: this.path } })
    }
  }

  /**
   * Get infile content.
   *
   * @public
   *
   * @return {string} Changelog infile content
   */
  public toString(): string {
    return fs.readFileSync(this.path, 'utf8')
  }
}

export default ChangelogInfile
