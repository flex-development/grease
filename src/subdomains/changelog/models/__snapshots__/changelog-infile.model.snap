// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:changelog/models/ChangelogInfile > #toString > should return infile content 1`] = `
"## [grease@1.0.2](https://github.com/flex-development/grease/compare/grease@1.0.1...grease@1.0.2) (2021-08-23)

### :pencil: Documentation

- [[\`6630524\`](https://github.com/flex-development/grease/commit/6630524c727da612482c1d2f0e152c44b3b1ee42)] **node:** update \`package.json#homepage\`
- [[\`32699c2\`](https://github.com/flex-development/grease/commit/32699c2a6bc05a06699c84bddb40f2b4fb191972)] **node:** update installation instructions

## [grease@1.0.1](https://github.com/flex-development/grease/compare/grease@1.0.0...grease@1.0.1) (2021-08-23)

### :robot: Continuous Integration

- [[\`ca0ae3d\`](https://github.com/flex-development/grease/commit/ca0ae3de3f76fd80043a52a06f1f97b3f74ea0ac)] **workflows:** add \`automate-pr-reviews\` workflow
- [[\`e331c4b\`](https://github.com/flex-development/grease/commit/e331c4b2199fa19bd79dfa6b48a9d498b3128129)] **workflows:** fix \`continuous-integration\` workflow
- [[\`2181e37\`](https://github.com/flex-development/grease/commit/2181e37bf3c64db510f3948a07bebceb4c475641)] **workflows:** update job secrets and environment variables
- [[\`1ceee8c\`](https://github.com/flex-development/grease/commit/1ceee8cfaf53a2002d5c990a5f6ae1156fb6bd42)] **workflows:** update workflow names

### :bug: Fixes

- [[\`e8ede43\`](https://github.com/flex-development/grease/commit/e8ede4365c7194e3f73d05413fdcbc2fecd037b7)] **node:** \`CreateReleaseDTO#version\` failure when \`options.skip.tag\` is \`true\`
- [[\`b89a932\`](https://github.com/flex-development/grease/commit/b89a932616f31547af5b642a81c9c139072cafdb)] **scripts:** \`releaseCommitMessageFormat\`

### :house_with_garden: Housekeeping

- [[\`3affb38\`](https://github.com/flex-development/grease/commit/3affb38bb20dd46d938c5ea8f4625ddf658d855e)] update release workflow
- [[\`9c96406\`](https://github.com/flex-development/grease/commit/9c9640604c2eb2d65d28dd3d36fce3c7538cce18)] update yarn2 config
- [[\`e99a18b\`](https://github.com/flex-development/grease/commit/e99a18b129fc0aaea089dee2e70961879dccd963)] **typescript:** update \`tsconfig.test.json\`

### :white_check_mark: Testing

- [[\`b1183f1\`](https://github.com/flex-development/grease/commit/b1183f1ca1812e3f94cf442ead84944a6aef8e54)] **node:** update \`IsSemVer\` functional test for version bump

## grease@1.0.0 (2021-08-15)

### :package: Build

- [[\`d7dd980\`](https://github.com/flex-development/grease/commit/d7dd9804da2da2e702e4535dbb2c39588981f8e4)] **deps-dev:** bump \`eslint-plugin-jsdoc\` to \`36.0.6\`
- [[\`9154d92\`](https://github.com/flex-development/grease/commit/9154d924ed1f081deaf913dfdef91bc74c37b792)] **deps-dev:** remove extraneous dependency \`ts-loader\`
- [[\`60789a6\`](https://github.com/flex-development/grease/commit/60789a6be855b4d81bdb951d1a3b05b46eb689df)] **deps:** bump @types/debug from 4.1.5 to 4.1.7 ([#1](https://github.com/flex-development/grease/issues/1))
- [[\`6f0d77a\`](https://github.com/flex-development/grease/commit/6f0d77a77ab56048a1c692900d10c666c8cad071)] **deps:** bump @types/lodash from 4.14.170 to 4.14.171 ([#3](https://github.com/flex-development/grease/issues/3))
- [[\`8504d8b\`](https://github.com/flex-development/grease/commit/8504d8be10064d89743c6de2af12dca9a2664265)] **deps:** bump @types/lodash from 4.14.171 to 4.14.172 ([#6](https://github.com/flex-development/grease/issues/6))
- [[\`8bbc133\`](https://github.com/flex-development/grease/commit/8bbc133947d49b550866fa72ff14a9199569e0fe)] **deps:** bump chalk from 4.1.1 to 4.1.2 ([#2](https://github.com/flex-development/grease/issues/2))
- [[\`8e5ec25\`](https://github.com/flex-development/grease/commit/8e5ec256d72e000e1f6d417caaedc12bd8391467)] **node:** add types only production build

### :pencil: Documentation

- [[\`857f9b0\`](https://github.com/flex-development/grease/commit/857f9b01333fe9b5c8d5a7b69bb4a4d6f50fd9c5)] code reviews
- [[\`a8fff2f\`](https://github.com/flex-development/grease/commit/a8fff2f748de5773d2eeb43ebef8fcb2b2804b0e)] fix link to \\\\\\\\\\"Generating Release Notes\\\\\\\\\\" section
- [[\`e74300f\`](https://github.com/flex-development/grease/commit/e74300f0bc933b992170a044390ca337497aecf5)] releasing
- [[\`fa65fde\`](https://github.com/flex-development/grease/commit/fa65fde247ee9e709e2da7a42a59e6ef0fa1a690)] update \`Getting Started\` section in \`README\`
- [[\`b0ad929\`](https://github.com/flex-development/grease/commit/b0ad929ece55c29d6b33521c39c0ecdb0a00f8b6)] update contributing guide
- [[\`73f2f1c\`](https://github.com/flex-development/grease/commit/73f2f1c31b6749f73fbb94941aae52d356d058b1)] update usage examples

### :sparkles: Features

- [[\`2e5894a\`](https://github.com/flex-development/grease/commit/2e5894a273d809a1f595a3e318f4e99656389f0c)] **node:** \`GreaseOptions.gitdir\`
- [[\`2b697aa\`](https://github.com/flex-development/grease/commit/2b697aada7bbb6c5f593f6ac2e32ee187b760700)] **node:** \`GreaseOptions.verify\`
- [[\`105ec5b\`](https://github.com/flex-development/grease/commit/105ec5b5a95b545d4dae9a2e5accc00b95f1494c)] **node:** add \`GreaseScripts.postrelease\`
- [[\`fc61afa\`](https://github.com/flex-development/grease/commit/fc61afacf4cb74d6c5173d7c97a0a82174a7ef56)] **node:** release branch whitelist
- [[\`38bedbc\`](https://github.com/flex-development/grease/commit/38bedbcaca0c84812519b5c65a6f8f6802c7b50c)] **node:** validate release data during \`greaser\` lifecycle

### :bug: Fixes

- [[\`8e65403\`](https://github.com/flex-development/grease/commit/8e65403f2290ad6269d8f36d3df9599e85f7d070)] \`commitlint.config\`
- [[\`5d34043\`](https://github.com/flex-development/grease/commit/5d34043d3a286673246f67789301fa7fbb30d100)] eslint typescript project settings
- [[\`fbe5265\`](https://github.com/flex-development/grease/commit/fbe5265945f0ff1b986c1a05486f7a2a8014a6cc)] **node:** \`cannot read property 'options' of undefined\`
- [[\`20abaf1\`](https://github.com/flex-development/grease/commit/20abaf1c01c3e9101d7831d1679aede152e8c4ae)] **node:** \`CreateReleaseDTO validation failure: [files,version]\`
- [[\`9cd322c\`](https://github.com/flex-development/grease/commit/9cd322c035c7e50e13b7d8a89c20d8cc21feaa8d)] **node:** \`CreateReleaseDTO validation failure: [version]\`
- [[\`a1a6f86\`](https://github.com/flex-development/grease/commit/a1a6f863f8185dbcdcf2ab8fc26f546b0ed1c45d)] **node:** \`CreateReleaseDTO#toString\`
- [[\`4dfdbe4\`](https://github.com/flex-development/grease/commit/4dfdbe4ee1be6bf0bca4485ff66e341a5921abd0)] **node:** \`CreateReleaseDTO#toString\`
- [[\`622d9ac\`](https://github.com/flex-development/grease/commit/622d9aca2c5044ff3cfc9ed1141b660b1787be34)] **node:** \`CreateReleaseDTO#toString\`
- [[\`4f90f0b\`](https://github.com/flex-development/grease/commit/4f90f0be1e348bf230ea34e1cc9a1191ece93fdc)] **node:** \`grease/main\` runs \`options.scripts.prerelease\` twice
- [[\`c6c9db0\`](https://github.com/flex-development/grease/commit/c6c9db0e4b44a1694dbb16327ad8451485c58d21)] **node:** \`grease/main\` runs \`options.scripts.prerelease\` twice
- [[\`fd8aece\`](https://github.com/flex-development/grease/commit/fd8aeceda095c05708c4b62359b1c04ef1c1e7ef)] **node:** \`GreaseOptions.infile\` decorator
- [[\`0a82c6e\`](https://github.com/flex-development/grease/commit/0a82c6e296d933fe6aefc3067ea922c1e59a0b0a)] **node:** \`IsTargetBranchConstraint.validate\`
- [[\`a49eb3b\`](https://github.com/flex-development/grease/commit/a49eb3bdca948b6c659f1d8329da6e85410fd807)] **node:** \`package.json#types\`
- [[\`36daabb\`](https://github.com/flex-development/grease/commit/36daabbbd1318e3c8fc057c6e4e37c17dd6e102e)] **node:** \`standard-version/lib/run-lifecycle-script\` usage
- [[\`68ccc1f\`](https://github.com/flex-development/grease/commit/68ccc1fab3f2c5423b29fe452a7a26f6e44bf158)] **node:** \`updaters_1.default is not a function\`
- [[\`8b3b8f4\`](https://github.com/flex-development/grease/commit/8b3b8f49cc58fab46b6059a1b3c2c9928e7a1786)] **node:** add \`GreaseScripts.postdepchecker\`
- [[\`91f560a\`](https://github.com/flex-development/grease/commit/91f560a5b00474f614831e451f069b04ebc495a1)] **node:** release tagging
- [[\`eff36a2\`](https://github.com/flex-development/grease/commit/eff36a28f58f0df47e7d743f673473870a252d02)] **scripts:** \`check:style\`
- [[\`3febb13\`](https://github.com/flex-development/grease/commit/3febb13dd383214d240f118839fc7d0052cd3098)] **typescript:** \`@zerollup/ts-transform-paths\` usage

### :house_with_garden: Housekeeping

- [[\`d3e9f8a\`](https://github.com/flex-development/grease/commit/d3e9f8ae00bdddd11f64849cfe9a20ec7bb872c0)] \`yarn set version 3.0.1\`
- [[\`34a51ba\`](https://github.com/flex-development/grease/commit/34a51ba8be5ef5822eaf34ea9d403122c9c1d56b)] add \`CHANGELOG\` files to \`.prettierignore\`
- [[\`3bbf8be\`](https://github.com/flex-development/grease/commit/3bbf8be104c85f30ee0674af96b252761a170018)] add \`publish:node\` script
- [[\`3d88b50\`](https://github.com/flex-development/grease/commit/3d88b50c95996d3b8794219b4ef2025645066fc5)] add \`scope:cache\` label
- [[\`2f29aa3\`](https://github.com/flex-development/grease/commit/2f29aa3698cf31043389f43502b7423c2dd106af)] P010-1 initial commit
- [[\`25ac6a6\`](https://github.com/flex-development/grease/commit/25ac6a6bb0d0c7569c33200a614a03c523ed9f8c)] P010-3 #done #time 15m
- [[\`1d609e9\`](https://github.com/flex-development/grease/commit/1d609e9ac2a411106a1eb425c683b257de71fc3e)] replace \`scope:grease-cli\` with \`scope:cli\`
- [[\`0e63fc7\`](https://github.com/flex-development/grease/commit/0e63fc7a1448d0d91dcee5243e8286feb119f5ff)] replace \`scope:grease\` label with \`scope:node\`
- [[\`0bd0082\`](https://github.com/flex-development/grease/commit/0bd0082a6e9f7a664f7e76969422b00ae108d721)] run \`yarn bootstrap\`
- [[\`127aea5\`](https://github.com/flex-development/grease/commit/127aea5c5131f8b40042d4a10c4b07a79e7c3ea2)] update publish config
- [[\`796873b\`](https://github.com/flex-development/grease/commit/796873b1343c11eaabd68a7a7eeb50759ff56314)] **node-release:** allow users to download package from github
- [[\`d15c166\`](https://github.com/flex-development/grease/commit/d15c166d75c60208b690e4d829c4d6065d2c62d3)] **node-scripts:** add release workflow script
- [[\`eb097c0\`](https://github.com/flex-development/grease/commit/eb097c0979a136892b6d2ac20d82c4ae5895be10)] **node-scripts:** update release workflow
- [[\`78a4b6a\`](https://github.com/flex-development/grease/commit/78a4b6ab20dfd8f1f37fd87e1287370e97cc23c7)] **node-scripts:** update release workflow
- [[\`92c75a1\`](https://github.com/flex-development/grease/commit/92c75a157935246168ac8314c722dd24bf60dcd5)] **node:** add \`prepublish\` script
- [[\`800d232\`](https://github.com/flex-development/grease/commit/800d2329212633c3974a9ec1b5bf33fda929fc34)] **node:** update \`package.json#publishConfig\`
- [[\`0de83c3\`](https://github.com/flex-development/grease/commit/0de83c3fdf156d683fd8c5ff4519c741b536c6e5)] **release:** @flex-development/grease@1.0.0

### :mechanical_arm: Refactors

- [[\`2267cb8\`](https://github.com/flex-development/grease/commit/2267cb8126e205b2b2955313d430e5144830a681)] implement monorepo architecture
- [[\`eddf14f\`](https://github.com/flex-development/grease/commit/eddf14fa2e68a9bfc9fe57ab5b541010f12866b5)] **node:** \`CreateReleaseDTO.files\` validation options
- [[\`7387e1a\`](https://github.com/flex-development/grease/commit/7387e1abff4f1d823cce80e5c44f6a95cfaa894b)] **node:** \`grease/main\` error log
- [[\`99a20e5\`](https://github.com/flex-development/grease/commit/99a20e5fc3ba39b15ebd7c1880c408bd0c5c2c25)] **node:** \`greaser\` lifecycle logging
- [[\`c319f84\`](https://github.com/flex-development/grease/commit/c319f840cff16462260e17f54c8862c13408b56c)] **node:** lifecycle logs
- [[\`c3bb103\`](https://github.com/flex-development/grease/commit/c3bb1038064468a01b2503b950a4644331256e82)] **node:** validation api

### :wastebasket: Reverts

- [[\`f56dea7\`](https://github.com/flex-development/grease/commit/f56dea7be7c0ab10c37972a76710b0a27362f9cd)] **node:** add types only production build
"
`;
